### PR TITLE
Health branch

### PR DESCRIPTION
**Pull request by lbassil**

### CODE REVIEW

File: transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py [hunk 1]
```diff
@@ -12,11 +12,13 @@
...
 import plotly.express as px
 from plotly.subplots import make_subplots
+import requests
+from io import StringIO
 
 
 mapbox_access_token = "pk.eyJ1IjoiamNyYW53ZWxsd2FyZCIsImEiOiJja2NkMW02aXcwYTl5MnFwbjdtdDB0M3oyIn0.zkIzPc4NSjLZvrY-DWrlZg"
 
-sdmx_url = "https://sdmx.data.unicef.org/ws/public/sdmxapi/rest/data/ECARO,TRANSMONEE,1.0/.{}....?format=csv"
+sdmx_url = "https://sdmx.data.unicef.org/ws/public/sdmxapi/rest/data/ECARO,TRANSMONEE,1.0/.{}...."
```
Comment (jamescw) [inline transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py:21]: Why did we drop csv format from this line?

File: transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py [hunk 2]
```diff
@@ -521,16 +631,30 @@ def geocode_address(address):
...
+    "endPeriod": str(years[-1]),
+}
+compress_header = {"Accept-Encoding": "gzip"}
+
 # avoid a loop to query SDMX
 try:
-    sdmx = pd.read_csv(sdmx_url.format("+".join(inds)), dtype=col_types)
+    response = requests.get(
+        sdmx_url.format("+".join(inds)), params=api_params, headers=compress_header
+    )
```
Comment (jamescw) [inline transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py:651]: why are we making a requests call here when `pd.read_csv` will do this already?

File: transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py [hunk 3]
```diff
@@ -521,16 +631,30 @@ def geocode_address(address):
...
 except urllib.error.HTTPError as e:
     raise e
 
+# bring requests response into pandas
+sdmx = pd.read_csv(StringIO(response.text), dtype=col_types)
```
Comment (jamescw) [inline transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py:656]: This is redundant,`read_csv` see above

File: transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py (hunk 2 above)
Comment (jamescw) [reply transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py:651]: I think I see why mostly this is for inclusion of the headers, pandas can already accommodate this: 

`sdmx_url = f"https://sdmx.data.unicef.org/ws/public/sdmxapi/rest/data/ECARO,TRANSMONEE,1.0/.{'+'.join(inds)}....?format=csv&startPeriod={years[0]}&endPeriod={years[-1]}"`

`pd.read_csv(sdmx_url, dtype=col_types, storage_options={"Accept-Encoding": "gzip"})`

This is documented here: https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html

Comment (jamescw) [reply transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py:651]: @beto-Sibileau take a look at this

File: transmonee_dashboard/src/transmonee_dashboard/settings.py [hunk 4]
```diff
@@ -43,6 +43,7 @@
         "rel": "preconnect",
     },
     "https://fonts.googleapis.com/css2?family=Roboto:wght@400;700&display=swap",
+    "https://codepen.io/chriddyp/pen/brPBPO.css",
```
Comment (jamescw) [inline transmonee_dashboard/src/transmonee_dashboard/settings.py:46]: Remove this for now @lbassil

**Review by jamescw — CHANGES_REQUESTED** [review]
Reviewed and suggested more concise way to pass headers to sdmx call

File: transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py (hunk 2 above)
Comment (beto-Sibileau) [reply transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py:651]: @jamescw thank you James, well noted, haven't thought about it before... I'll take a look at the documentation, maybe we could add the parameters format, startPeriod, endPeriod as well, would make the url more clear (not very relevant, just getting picky =)

**Review by beto-Sibileau — APPROVED** [review]

File: transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py (hunk 2 above)
Comment (beto-Sibileau) [reply transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py:645]: > @beto-Sibileau take a look at this

@james any clue on why this is not being considered as a HTTPS url?

```
sdmx_url = "https://sdmx.data.unicef.org/ws/public/sdmxapi/rest/data/ECARO,TRANSMONEE,1.0/.{}....?format=csv&startPeriod={}&endPeriod={}"
sdmx = pd.read_csv(
    sdmx_url.format("+".join(inds), years[0], years[-1]),
    dtype=col_types,
    storage_options={"Accept-Encoding": "gzip"},
    )
```

The error I got looks to me related to the fact that the URL is being handled by fsspec

`ValueError: storage_options passed with file object or non-fsspec file path`


Pandas doc:

image

**Review by beto-Sibileau — APPROVED** [review]

**Review by beto-Sibileau — APPROVED** [review]

**Review by beto-Sibileau — APPROVED** [review]

**Review by beto-Sibileau — APPROVED** [review]

File: transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py (hunk 2 above)
Comment (beto-Sibileau) [reply transmonee_dashboard/src/transmonee_dashboard/pages/__init__.py:645]: LGTM, only issue to add is the headers

**Review by beto-Sibileau — APPROVED** [review]

**Review by jamescw — APPROVED** [review]
I have fixed the storage_options problem, it was related to the version of Pandas which I have now pinned at 1.3.0 @beto-Sibileau please test locally then merge this PR if you are happy

**Review by beto-Sibileau — APPROVED** [review]
LGTM

**Pull request merged**